packages/web-components/@sfx/product/src/product.ts

```diff
@@ -0,0 +1,104 @@
+import { customElement, property, html, TemplateResult } from 'lit-element';
+
+import { Base } from '@sfx/base';
+
+@customElement('sfx-product')
+export default class Product extends Base {
+  @property({ type: Object }) product: ProductModel | any = {};
+
+  constructor() {
+    super();
+    this.updateVariant = this.updateVariant.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener('sfx::change_product_variant', this.updateVariant);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener('sfx::change_product_variant', this.updateVariant);
+  }
+
+  updateVariant(e: CustomEvent) {
+    const product = e.detail;
+
+    this.product = {
+      ...this.product,
+      ...product
+    };
+  }
+
+  urlWrap(url: string, children: TemplateResult) {
+    return url ? html`<a href="${ url }">${children}</a>` : html`${children}`;
```

Comment:
If `children` is already a `TemplateResult`, wouldn't it be better to simply return it in the case where the `url` is falsy?